average.js

```diff
@@ -1,6 +1,22 @@
+function average(numbers){
+    if(numbers.length == 0){
+        return NaN;
+    }
+
+    var result = numbers.filter(function(number){
+        if(!isNaN(number) && (number > -50 && number < 80)){
```

Comment:
1. Please mention the rationale for choosing 80 as the limit of temperature
2. This condition will keep getting complex during enhancements: e.g., addition of some batteries that work in high temperature environments. Can you think of a way to limit the complexity growth?

Reply:
1. As I was going EV battery temperature ranges I came through a graph in the post [https://www.geotab.com/blog/ev-range/](url) the graph shows that if the temperature goes above 40 the range goes to 60% of its max range. SO I think ranges above that should be outliers.

Reply:
2. we can dynamically compute the outliers using the interquartile method.

Reply:
I think changing the outlier detection and making it dynamic will do the work as we do in machine learning.

Reply:
But detecting outlier will come with its own set of problems as in the IQR method we have to sort the array which could be problematic if the size is large and the Standard Deviation method would require the data to be in Gaussian distribution but it will not require sorting of the data.

Reply:
@sudeeprp @numocityadmin I think the standard deviation solution would be great for this, hoping to hear your input on this.

Reply:
@sudeeprp @numocityadmin I have the code for standard deviation too, I think this is the most efficient way to do this as sorting is not required in this case and if the array gets big(as it will in production) it won't affect the performance. I hope you like the idea.